packages/block-editor/src/components/block-list/block-wrapper.js

```diff
@@ -185,37 +191,82 @@ const BlockComponent = forwardRef(
 			if ( ( buttons || which ) === 1 ) {
 				onSelectionStart( clientId );
 			}
+		}
+
+		if ( isSelected ) {
+			// Only allow shortcuts when a blocks is selected and not locked.
+			if ( ! isLocked ) {
+				ref.current.addEventListener( 'keydown', onKeyDown );
+			}
+
+			// Only allow selection to be started from a selected block.
+			ref.current.addEventListener( 'mouseleave', onMouseLeave );
+		}
+
+		return () => {
+			if ( isSelected ) {
+				if ( ! isLocked ) {
+					ref.current.removeEventListener( 'keydown', onKeyDown );
+				}
+
+				ref.current.removeEventListener( 'mouseleave', onMouseLeave );
+			}
 		};
+	}, [
+		isSelected,
+		isLocked,
+		insertDefaultBlock,
+		removeBlock,
+		onSelectionStart,
+	] );
 
-		const htmlSuffix =
-			mode === 'html' && ! __unstableIsHtml ? '-visual' : '';
-		const blockElementId = `block-${ clientId }${ htmlSuffix }`;
+	const htmlSuffix = mode === 'html' && ! __unstableIsHtml ? '-visual' : '';
 
+	return {
+		...wrapperProps,
+		...props,
+		ref,
+		id: `block-${ clientId }${ htmlSuffix }`,
+		tabIndex: 0,
+		role: 'group',
+		'aria-label': blockLabel,
+		'data-block': clientId,
+		'data-type': name,
+		'data-title': blockTitle,
+		className: classnames(
+			'wp-block',
+			className,
+			props.className,
+			wrapperProps.className
+		),
+		style: { ...wrapperProps.style, ...props.style },
+	};
+}
+
+const BlockComponent = forwardRef(
+	(
+		{ children, tagName: TagName = 'div', __unstableIsHtml, ...props },
+		wrapper
+	) => {
+		const wrapperProps = useBlockProps(
+			{ ref: wrapper },
+			{ __unstableIsHtml }
+		);
+		const isAligned = wrapperProps && !! wrapperProps[ 'data-align' ];
```

Comment:
There's still a lot of things remaining here. alignment support, labels... role. Can't we absorb everything?